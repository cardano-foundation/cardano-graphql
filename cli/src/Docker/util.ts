import { QuestionCollection } from 'inquirer'
import { DockerComposeStack, StatefulService } from './index'

export function selectServicesFromStack (stack: DockerComposeStack, multi = true): QuestionCollection {
  const label = multi === true ? 'services' : 'service'
  return {
    name: label,
    message: `Select ${label}`,
    type: multi === true ? 'checkbox' : 'list',
    choices: [...stack.statefulServices.values()].map(service => {
      return {
        name: service.name,
        value: service
      }
    }),
    validate (input) {
      if (input.length === 0) {
        throw new Error('At least one service must be selected')
      }
      return true
    }
  }
}

export function servicesFromNameArg (stack: DockerComposeStack, names: string): StatefulService[] {
  return names.split(',').map((name: string) => {
    if (!stack.statefulServices.has(name)) {
      throw new Error(`Unknown service ${name}`)
    }
    return stack.statefulServices.get(name)
  })
}
