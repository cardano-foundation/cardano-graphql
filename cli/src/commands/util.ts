import { DockerComposeStack } from '../docker'

export function serviceChoices (stack: DockerComposeStack) {
  return [...stack.statefulServices.values()].map(service => {
    return { name: service.name, value: service }
  })
}

export function serviceFromName (stack: DockerComposeStack, name: string) {
  if (!stack.statefulServices.has(name)) {
    throw new Error(`Unknown service ${name}`)
  }
  return stack.statefulServices.get(name)
}

export function servicesFromNames (stack: DockerComposeStack, names: string) {
  return names.split(',').map((name: string) => serviceFromName(stack, name))
}
