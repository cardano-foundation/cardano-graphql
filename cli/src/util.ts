
export function ensureValue (input: string) {
  return !input ? 'Cannot be blank' : true
}
