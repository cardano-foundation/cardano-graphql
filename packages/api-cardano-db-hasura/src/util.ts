import { ConnectionConfig, createInteractionContext } from '@cardano-ogmios/client'
import { Logger } from 'ts-log'

export const createInteractionContextWithLogger = (connection: ConnectionConfig, logger: Logger, module: string, onClose?: () => Promise<void>) =>
  createInteractionContext(
    (error) => {
      logger.error({ module, error }, error.message)
    },
    async (code, reason) => {
      if (code === 1000) {
        logger.info({ module, code }, reason)
      } else {
        logger.error({ module, code }, 'Connection closed')
        await onClose?.()
      }
    },
    {
      connection
    }
  )
