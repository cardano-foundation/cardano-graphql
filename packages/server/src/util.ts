import { Application } from 'express'
import http from 'http'

export function listenPromise (app: Application, port: number, hostname?: string): Promise<http.Server> {
  return new Promise(function (resolve, reject) {
    const server: http.Server = app.listen(port, hostname, () => resolve(server))
    server.on('error', reject)
  })
}
