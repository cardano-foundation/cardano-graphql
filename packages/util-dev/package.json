{
  "name": "@cardano-graphql/util-dev",
  "version": "1.0.0",
  "description": "Common development utilities",
  "main": "dist/index.js",
  "typings": "dist/index.d.ts",
  "scripts": {
    "build": "tsc -b ./src",
    "cleanup": "shx rm -rf dist node_modules",
    "lint": "eslint --ignore-path ../../.eslintignore \"**/*.ts\"",
    "prepack": "yarn build",
    "test": "shx echo No tests in this package"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/input-output-hk/cardano-graphql.git"
  },
  "author": "Rhys Bartels-Waller",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/input-output-hk/cardano-graphql/issues"
  },
  "homepage": "https://github.com/input-output-hk/cardano-graphql/blob/master/packages/util-dev/README.md",
  "devDependencies": {
    "@cardano-graphql/util": "1.0.0",
    "shx": "^0.3.2"
  },
  "directories": {
    "lib": "src",
    "test": "test"
  },
  "files": [
    "dist",
    "LICENSE",
    "NOTICE",
    "README.md"
  ]
}
