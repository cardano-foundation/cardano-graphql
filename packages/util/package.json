{
  "name": "@cardano-graphql/util",
  "version": "1.0.0",
  "description": "Common utilities",
  "main": "dist/index.js",
  "typings": "dist/index.d.ts",
  "scripts": {
    "build": "tsc -b ./src",
    "cleanup": "shx rm -rf dist node_modules",
    "lint": "eslint --ignore-path ../../.eslintignore \"**/*.ts\"",
    "prepack": "yarn build",
    "test": "shx echo No tests in this package"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/input-output-hk/cardano-graphql.git"
  },
  "author": "Rhys Bartels-Waller",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/input-output-hk/cardano-graphql/issues"
  },
  "homepage": "https://github.com/input-output-hk/cardano-graphql/blob/master/packages/util/README.md",
  "dependencies": {
    "graphql": "14.5.8",
    "graphql-bigint": "^1.0.0",
    "graphql-tag": "^2.10.3",
    "p-retry": "^4.2.0"
  },
  "devDependencies": {
    "@types/node": "^14.0.13",
    "shx": "^0.3.2",
    "typescript": "^3.9.5"
  },
  "directories": {
    "lib": "src",
    "test": "test"
  },
  "files": [
    "dist",
    "LICENSE",
    "NOTICE",
    "README.md"
  ]
}
