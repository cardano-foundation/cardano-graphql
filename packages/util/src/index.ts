import { onFailedAttemptFor } from './onFailedAttemptFor'
import { loadQueryNode } from './queryNodeLoading'
import * as dateTimeValidation from './dateTimeValidation'
import * as scalars from './scalars'

export default {
  dateTimeValidation,
  onFailedAttemptFor,
  loadQueryNode,
  scalars
}
