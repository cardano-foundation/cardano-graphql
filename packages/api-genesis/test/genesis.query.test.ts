import path from 'path'
import { DocumentNode } from 'graphql'
import util from '@cardano-graphql/util'
import { TestClient } from '@cardano-graphql/util-dev'
import { buildClient } from './util'

const shelleyTestnetGenesis = require('../../../config/network/shelley_testnet/genesis_shelley.json')
const mainnetCandidateGenesis = require('../../../config/network/mainnet_candidate/genesis_shelley.json')

function loadQueryNode (name: string): Promise<DocumentNode> {
  return util.loadQueryNode(path.resolve(__dirname, '..', 'src', 'example_queries'), name)
}

describe('genesis', () => {
  let shelleyTestnetClient: TestClient
  let mainnetCandidateClient: TestClient

  describe('Shelley testnet', () => {
    beforeAll(async () => {
      shelleyTestnetClient = await buildClient(shelleyTestnetGenesis)
      mainnetCandidateClient = await buildClient(mainnetCandidateGenesis)
    })

    it('Returns key information about the network', async () => {
      const query = { query: await loadQueryNode('keyNetworkInfo') }
      const shelleyTestnetResult = await shelleyTestnetClient.query(query)
      expect(shelleyTestnetResult.data).toMatchSnapshot()
      const mainnetCandidateResult = await mainnetCandidateClient.query(query)
      expect(mainnetCandidateResult.data).toMatchSnapshot()
    })

    it('Returns all information from genesis.json', async () => {
      const query = { query: await loadQueryNode('allInfo') }
      const shelleyTestnetResult = await shelleyTestnetClient.query(query)
      expect(shelleyTestnetResult.data).toMatchSnapshot()
      const mainnetCandidateResult = await mainnetCandidateClient.query(query)
      expect(mainnetCandidateResult.data).toMatchSnapshot()
    })
  })
})
