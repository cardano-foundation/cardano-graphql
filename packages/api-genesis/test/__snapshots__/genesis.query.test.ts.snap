// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genesis Shelley testnet Returns all information from genesis.json 1`] = `
Object {
  "genesis": Object {
    "shelley": Object {
      "activeSlotsCoeff": 0.05,
      "epochLength": 21600,
      "genDelegs": Object {
        "46accc160a32598fd87b14d823fd25ec6ca3388316c404ce5aa06dca": Object {
          "delegate": "e47b777efea3146b1aa18fde95ddbd3de38f068f529997ff4141e484",
          "vrf": "56c2364bb4a90d0f674ae1b758d5c19b2f1622b3f763799f1f0793cfc4591daf",
        },
        "c800db9af0cfe4997163e730ff2eeadc984f4ed29bdfeed37fb87c05": Object {
          "delegate": "252a3464a2e6d3e824e3dc63629276838227f92f2c6b6cc45fc16138",
          "vrf": "ca23e0093e70086bf92788f0efcbe45c1727f19d4376a94118d5bfaebdcca257",
        },
        "cbdf7c87a498440cc16fb16b6d3a957de7979252de695e38344aee52": Object {
          "delegate": "609114bdfa04f8fa2dc3450306427a6bb803a1cce84a2c3c760a7e4b",
          "vrf": "48a39ab96ed61faa76b69a08f39282e6a4fb458020e024ee6b612c93f4439c99",
        },
      },
      "initialFunds": Object {
        "6034d2c8b6373a4a3326259c1fbccfc45405bd20552097d960868b40a9": 10000000000000000,
        "6051d08fc78ca26cfc41b839f6c30b25ccf45f81594d4147a6f392a058": 1000000000000000,
      },
      "maxKESEvolutions": 120,
      "maxLovelaceSupply": "45000000000000000",
      "networkId": "Testnet",
      "networkMagic": 42,
      "protocolMagicId": 42,
      "protocolParams": Object {
        "a0": 0.3,
        "decentralisationParam": 1,
        "eMax": 18,
        "extraEntropy": Object {
          "tag": "NeutralNonce",
        },
        "keyDeposit": 400000,
        "maxBlockBodySize": 65536,
        "maxBlockHeaderSize": 1100,
        "maxTxSize": 16384,
        "minFeeA": 44,
        "minFeeB": 155381,
        "minPoolCost": 0,
        "minUTxOValue": 0,
        "nOpt": 250,
        "poolDeposit": 500000000,
        "protocolVersion": Object {
          "major": 0,
          "minor": 0,
        },
        "rho": 0.0022,
        "tau": 0.05,
      },
      "securityParam": 108,
      "slotLength": 1,
      "slotsPerKESPeriod": 3600,
      "staking": Object {
        "pools": Object {},
        "stake": Object {},
      },
      "systemStart": "2020-07-13T11:00:00Z",
      "updateQuorum": 3,
    },
  },
}
`;

exports[`genesis Shelley testnet Returns all information from genesis.json 2`] = `
Object {
  "genesis": Object {
    "shelley": Object {
      "activeSlotsCoeff": 0.05,
      "epochLength": 21600,
      "genDelegs": Object {
        "46accc160a32598fd87b14d823fd25ec6ca3388316c404ce5aa06dca": Object {
          "delegate": "e47b777efea3146b1aa18fde95ddbd3de38f068f529997ff4141e484",
          "vrf": "56c2364bb4a90d0f674ae1b758d5c19b2f1622b3f763799f1f0793cfc4591daf",
        },
        "c800db9af0cfe4997163e730ff2eeadc984f4ed29bdfeed37fb87c05": Object {
          "delegate": "252a3464a2e6d3e824e3dc63629276838227f92f2c6b6cc45fc16138",
          "vrf": "ca23e0093e70086bf92788f0efcbe45c1727f19d4376a94118d5bfaebdcca257",
        },
        "cbdf7c87a498440cc16fb16b6d3a957de7979252de695e38344aee52": Object {
          "delegate": "609114bdfa04f8fa2dc3450306427a6bb803a1cce84a2c3c760a7e4b",
          "vrf": "48a39ab96ed61faa76b69a08f39282e6a4fb458020e024ee6b612c93f4439c99",
        },
      },
      "initialFunds": Object {
        "6034d2c8b6373a4a3326259c1fbccfc45405bd20552097d960868b40a9": 10000000000000000,
        "6051d08fc78ca26cfc41b839f6c30b25ccf45f81594d4147a6f392a058": 1000000000000000,
      },
      "maxKESEvolutions": 120,
      "maxLovelaceSupply": "45000000000000000",
      "networkId": "Testnet",
      "networkMagic": 42,
      "protocolMagicId": 42,
      "protocolParams": Object {
        "a0": 0.3,
        "decentralisationParam": 1,
        "eMax": 18,
        "extraEntropy": Object {
          "tag": "NeutralNonce",
        },
        "keyDeposit": 400000,
        "maxBlockBodySize": 65536,
        "maxBlockHeaderSize": 1100,
        "maxTxSize": 16384,
        "minFeeA": 44,
        "minFeeB": 155381,
        "minPoolCost": 0,
        "minUTxOValue": 0,
        "nOpt": 250,
        "poolDeposit": 500000000,
        "protocolVersion": Object {
          "major": 0,
          "minor": 0,
        },
        "rho": 0.0022,
        "tau": 0.05,
      },
      "securityParam": 108,
      "slotLength": 1,
      "slotsPerKESPeriod": 3600,
      "staking": Object {
        "pools": Object {},
        "stake": Object {},
      },
      "systemStart": "2020-07-13T11:00:00Z",
      "updateQuorum": 3,
    },
  },
}
`;

exports[`genesis Shelley testnet Returns key information about the network 1`] = `
Object {
  "genesis": Object {
    "shelley": Object {
      "epochLength": 21600,
      "maxLovelaceSupply": "45000000000000000",
      "networkId": "Testnet",
      "networkMagic": 42,
      "securityParam": 108,
      "slotLength": 1,
      "slotsPerKESPeriod": 3600,
      "systemStart": "2020-07-13T11:00:00Z",
    },
  },
}
`;

exports[`genesis Shelley testnet Returns key information about the network 2`] = `
Object {
  "genesis": Object {
    "shelley": Object {
      "epochLength": 21600,
      "maxLovelaceSupply": "45000000000000000",
      "networkId": "Testnet",
      "networkMagic": 42,
      "securityParam": 108,
      "slotLength": 1,
      "slotsPerKESPeriod": 3600,
      "systemStart": "2020-07-13T11:00:00Z",
    },
  },
}
`;
